### PR TITLE
Tiger910 keyboard initial commit

### PR DESCRIPTION
**Pull request by Weirdo-F**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/tiger910/keymaps/default/config.h [hunk 1]
```diff
@@ -0,0 +1,69 @@
+/*
```
Comment (zvecr) [inline keyboards/tiger910/keymaps/default/config.h:1]: This file should be moved to `keyboards/tiger910/config.h`

File: keyboards/tiger910/rules.mk [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# MCU name
+MCU = STM32F103
+
+# Bootloader selection
+BOOTLOADER = stm32duino
+
+# project specific files
+VPATH += keyboards/cannonkeys/bluepill
+SRC =	keyboard.c
```
Comment (zvecr) [inline keyboards/tiger910/rules.mk:9]:
```suggestion
```

File: keyboards/tiger910/keymaps/rev1/config.h [hunk 3]
```diff
@@ -0,0 +1,69 @@
+/*
```
Comment (zvecr) [inline keyboards/tiger910/keymaps/rev1/config.h:1]: This should be removed

File: keyboards/tiger910/keymaps/rev1/keymap.c [hunk 4]
```diff
@@ -0,0 +1,60 @@
+/*
```
Comment (zvecr) [inline keyboards/tiger910/keymaps/rev1/keymap.c:1]: Not sure this keymap adds value over default, so should be removed.

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/tiger910/keymaps/rev1/config.h (hunk 3 above)
Comment (Weirdo-F) [reply keyboards/tiger910/keymaps/rev1/config.h:1]: This version has changed the number of RGB lights and needs to modify the configuration

File: keyboards/tiger910/keymaps/rev1/keymap.c (hunk 4 above)
Comment (Weirdo-F) [reply keyboards/tiger910/keymaps/rev1/keymap.c:1]: This version changes the number of RGB lights, so you need to modify the RGB control statement.

File: keyboards/tiger910/keymaps/default/config.h (hunk 1 above)
Comment (Weirdo-F) [reply keyboards/tiger910/keymaps/default/config.h:1]: The two versions have different RGB number settings

File: keyboards/tiger910/keymaps/rev1/config.h (hunk 3 above)
Comment (drashna) [reply keyboards/tiger910/keymaps/rev1/config.h:1]: Specifically, this should not be in a keymap folder.  It should be in `/tiger910/rev1/` instead.

Comment (Weirdo-F) [reply keyboards/tiger910/keymaps/rev1/config.h:1]: > Specifically, this should not be in a keymap folder.  It should be in `/tiger910/rev1/` instead. 

I tried to do this, but I got compile errors

Comment (drashna) [reply keyboards/tiger910/keymaps/rev1/config.h:1]: That may be due to the naming and such.  If you post the errors, and push the code, we can take a look and help you fix those errors.

Comment (Weirdo-F) [reply keyboards/tiger910/keymaps/rev1/config.h]: OK.
I've made some changes.

File: keyboards/tiger910/rev1/tiger910.c [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  * 
+  * This program is distributed in the hope that it will be useful, 
+  * but WITHOUT ANY WARRANTY; without even the implied warranty of 
+  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the 
+  * GNU General Public License for more details. 
+  * 
+  * You should have received a copy of the GNU General Public License 
+  * along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+  */ 
+#include "tiger910.h"
```
Comment (drashna) [inline keyboards/tiger910/rev1/tiger910.c:16]: This file should be named `rev1.c`, and: 
```suggestion
#include "rev1.h"
```

File: keyboards/tiger910/rev1/tiger910.h [hunk 6]
```diff
@@ -0,0 +1,34 @@
+  /* Copyright 2020 Weirdo
```
Comment (drashna) [inline keyboards/tiger910/rev1/tiger910.h:1]: This file should be named `rev1.h`

File: keyboards/tiger910/rev2/tiger910.c [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  * 
+  * This program is distributed in the hope that it will be useful, 
+  * but WITHOUT ANY WARRANTY; without even the implied warranty of 
+  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the 
+  * GNU General Public License for more details. 
+  * 
+  * You should have received a copy of the GNU General Public License 
+  * along with this program.  If not, see <http://www.gnu.org/licenses/>. 
+  */ 
+#include "tiger910.h"
```
Comment (drashna) [inline keyboards/tiger910/rev2/tiger910.c:16]: This file should be named `rev2.c`, and:
```suggestion
#include "rev2.h"
```

File: keyboards/tiger910/rev2/tiger910.h [hunk 8]
```diff
@@ -0,0 +1,34 @@
+  /* Copyright 2020 Weirdo
```
Comment (drashna) [inline keyboards/tiger910/rev2/tiger910.h:1]: This file should be named `rev2.h`

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/tiger910/rev2/tiger910.h (hunk 8 above)
Comment (Weirdo-F) [reply keyboards/tiger910/rev2/tiger910.h]: I've nade these changes.

**Review by drashna — APPROVED** [review]

**Comment by Weirdo-F** [comment]
Are there any other errors that need to be corrected? @drashna

**Comment by drashna** [comment]
No, looks good to me, and have verified that it compiles. 

It's just waiting on a 2nd approval.

**Comment by Weirdo-F** [comment]
@drashna What can I do at this time?

File: keyboards/tiger910/rev1/chconf.h [hunk 9]
```diff
@@ -0,0 +1,714 @@
...
+ *
+ * @addtogroup config
+ * @details Kernel related settings and hooks.
+ * @{
+ */
+
+#ifndef CHCONF_H
+#define CHCONF_H
+
+#define _CHIBIOS_RT_CONF_
```
Comment (drashna) [inline keyboards/tiger910/rev1/chconf.h:31]: Could you cleanup this file?  

We don't use the version specific config anymore.  You can find the "defaults" to use here: 
https://github.com/qmk/qmk_firmware/tree/master/platforms/chibios/keyboard-config-templates

And then basically only enable what you need enabled.

File: keyboards/tiger910/rev1/halconf.h [hunk 10]
```diff
@@ -0,0 +1,525 @@
+/*
```
Comment (drashna) [inline keyboards/tiger910/rev1/halconf.h:1]: Same here.

File: keyboards/tiger910/rev1/mcuconf.h [hunk 11]
```diff
@@ -0,0 +1,209 @@
+/*
```
Comment (drashna) [inline keyboards/tiger910/rev1/mcuconf.h:1]: same here.

File: keyboards/tiger910/rev2/chconf.h [hunk 12]
```diff
@@ -0,0 +1,714 @@
+/*
```
Comment (drashna) [inline keyboards/tiger910/rev2/chconf.h:1]: Same here.

File: keyboards/tiger910/rev2/halconf.h [hunk 13]
```diff
@@ -0,0 +1,525 @@
+/*
```
Comment (drashna) [inline keyboards/tiger910/rev2/halconf.h:1]: same here.

File: keyboards/tiger910/rev2/mcuconf.h [hunk 14]
```diff
@@ -0,0 +1,209 @@
+/*
```
Comment (drashna) [inline keyboards/tiger910/rev2/mcuconf.h:1]: same

**Review by drashna — CHANGES_REQUESTED** [review]
Sorry, a few things have changed due to ChibiOS related upgrades. 

If you could update the chibiOS config files, then it should be good to go (and won't need updates in the future)

File: keyboards/tiger910/rev1/chconf.h (hunk 9 above)
Comment (Weirdo-F) [reply keyboards/tiger910/rev1/chconf.h:31]: I'm very sorry, I didn't understand how I should modify it.

**Comment by Weirdo-F** [comment]
> Sorry, a few things have changed due to ChibiOS related upgrades. 
> 
> If you could update the chibiOS config files, then it should be good to go (and won't need updates in the future)

Could you tell me in detail how to modify it? thanks！

**Comment by drashna** [comment]
Could you run `qmk chibios-confmigrate` for this board?

**Comment by Weirdo-F** [comment]
> qmk chibios-confmigrate

what is that？

**Pull request closed without merging**